src/main/java/seedu/address/logic/commands/DeleteStudentCommand.java

```diff
@@ -44,14 +71,56 @@ public DeleteStudentCommand(Name name) {
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
+        ObservableList<Student> studentList = model.getFilteredStudentList();
+        ArrayList<Student> listToCheck = new ArrayList<>();
+
+        for (Student student : studentList) {
+            if (student.getName().equals(name)) {
+                listToCheck.add(student);
+            }
+        }
+
+        if (listToCheck.isEmpty()) {
+            throw new CommandException(MESSAGE_NONEXISTENT_STUDENT);
+        }
+
+        // if there are multiple students with the same name, ensure that student number is provided
+        if (listToCheck.size() > 1 && this.studentNumber == null) {
+            StringBuilder duplicates = new StringBuilder();
+            for (Student student : listToCheck) {
+                duplicates.append(student.getStudentNumber());
+                duplicates.append(" ");
+            }
+            String duplicateStudentNumbers = duplicates.toString();
+            throw new CommandException(String.format(MESSAGE_DUPLICATE_STUDENT, duplicateStudentNumbers, name));
+        }
+
+        if (listToCheck.size() > 1) {
+            for (Student student : listToCheck) {
+                if (student.getStudentNumber().equals(studentNumber)) {
+                    model.deleteStudent(student);
+                    return new CommandResult(String.format(MESSAGE_DELETE_STUDENT_SUCCESS) + student.getName()
+                            + " " + student.getStudentNumber());
```

Comment:
Perhaps to keep the style consistent, you could use string format here as well like ur duplicate student, but this is not neccassary if you don't want to.